src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +66,249 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
+ * different size functions supported by PG are also supported by this function changing
+ * the size query given as the last parameter to function. Format of sizeQuery is
+ * pg_*_size(%s). Examples of it can be found in the master_protocol.h
+ */
+StringInfo
+GenerateSizeQueryOnMultiplePlacements(Oid distributedRelationId, List *shardIntervalList,
+									  char *sizeQuery)
+{
+	/* schema name will be used to form shard qualified name */
+	Oid schemaId = get_rel_namespace(distributedRelationId);
+	char *schemaName = get_namespace_name(schemaId);
+
+	StringInfo selectQuery = makeStringInfo();
+	ListCell *shardCell = NULL;
+
+	appendStringInfo(selectQuery, "SELECT ");
+
+	foreach(shardCell, shardIntervalList)
+	{
+		ShardInterval *shardInterval = (ShardInterval *) lfirst(shardCell);
+		uint64 shardId = shardInterval->shardId;
+		char *shardName = get_rel_name(distributedRelationId);
+		char *shardQualifiedName = NULL;
+		char *quotedShardName = NULL;
+		AppendShardIdToName(&shardName, shardId);
+
+		shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
+		quotedShardName = quote_literal_cstr(shardQualifiedName);
+
+		appendStringInfo(selectQuery, sizeQuery, quotedShardName);
+		appendStringInfo(selectQuery, " + ");
+	}
+
+	/* there is no more size function so just add 0 instead of it */
+	appendStringInfo(selectQuery, "0;");
+	return selectQuery;
+}
+
+
+/*
+ * GetDistributedTableSizeOnWorker gets the workerNode and relationId to calculate size
+ * of that relation on the given workerNode by summing up the size of each shard placement.
+ */
+uint64
+GetDistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId, char *sizeQuery)
+{
+	StringInfo tableSizeQuery = NULL;
+	char *workerNodeName = workerNode->workerName;
+	uint32 workerNodePort = workerNode->workerPort;
+	StringInfo tableSizeStringInfo;
+	char *tableSizeString;
+	List *sizeList = NIL;
+	uint64 relationSize = 0;
+	List *shardIntervalsOnNode = GetShardIntervalsOnWorkerNode(workerNode, relationId);
+
+	tableSizeQuery = GenerateSizeQueryOnMultiplePlacements(relationId,
+														   shardIntervalsOnNode,
+														   sizeQuery);
+
+	sizeList = ExecuteRemoteQuery(workerNodeName, workerNodePort, NULL, tableSizeQuery);
+
+	if (sizeList == NIL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
+						errmsg("cannot get the size")));
+	}
+
+	tableSizeStringInfo = (StringInfo) linitial(sizeList);
+	tableSizeString = tableSizeStringInfo->data;
+	relationSize = atol(tableSizeString);
+
+	return relationSize;
+}
+
+
+/*
+ * DistributedTableSize is a helper function for all citus size functions. It
+ * first checks whether the table is distributed and size query can be run on it.
+ * Connection to each node has to be established to get the size of the table.
+ */
+uint64
+DistributedTableSize(Oid relationId, char *sizeQuery)
+{
+	if (XactModificationLevel == XACT_MODIFICATION_MULTI_SHARD)
```

Comment:
Let's move this check after the variable declarations. However, to not calcualte WorkerNodeList in case of the error, first assign workerNodeList to NULL, after the if check(and even after the taking lock) you can assign it to WorkerNodeList.

Also there is no need to else block. ereport does not continue after error happens.